Homework 2

yasksc opened the pull request:
1) in homework-1 branch: 
added functionality to calculate current value for buy and sell transactions
2) in homework-2 branch:
added profit / loss calculation functionality for sales transactions

yasksc committed:
added functionality to calculate current value for buy and sell transactions

yasksc committed:
added profit / loss calculation functionality for sales transactions

yasksc commented:
for homework-1 branch I just forgot remove (+) from line 59 in CurrentValueCalc class:
CurrentValue += (int)trcType * Nominal * Price;
it should be like this:
CurrentValue = (int)trcType * Nominal * Price;

d-okhrimenko commented on Homework2/CurrentValueCalc.cs:
```diff
@@ -0,0 +1,84 @@
+
+namespace Homework2
+{
+    enum TransactionType
```
please move this enum type to a separate file

d-okhrimenko commented on Homework2/ICurrentValueCalc.cs:
```diff
@@ -0,0 +1,15 @@
+        int Nominal { get; set; }
+        double PriceTrade { get; set; }
+        double PriceOrigin { get; set; }
+        string TransType { get; set; }
```
If possible we should avoid abbrivaitions. I thik TransactionType name is better

d-okhrimenko commented on Homework2/CurrentValueCalc.cs:
```diff
@@ -0,0 +1,84 @@
+                }
+                else
+                {
+                    Console.WriteLine("Price must be bigger than 0");
```
In future it must be replased by exception

d-okhrimenko reviewed (changes requested):
Great job! 👍  but need minor improvments

yasksc committed:
1) added exceptions
2) Renamed TransType properties to TransactionTypeUser
3) added enum to a separate file

d-okhrimenko commented on Homework2/CurrentValueCalc.cs:
```diff
@@ -0,0 +1,97 @@
+                catch (Exception e)
+                {
+                    Console.WriteLine(e.Message);
+                }
```
try/catch must be moved to another location where CurrentValueCalc will be used. In this case, the class user can decide how exception should be handled and whether it should be a message to the console or a message box or anouther action, such as requesting another value from the user.

d-okhrimenko commented on Homework2/CurrentValueCalc.cs:
```diff
@@ -0,0 +1,97 @@
+            PriceTrade = priceTrade;
+            TransactionTypeUser = transactionTypeUser;
+        }
+        public void CurrentValueCalculation()
```
Try to use verbs for naming methods. For example, CurrentValueCalculation can be changed to CalculateCurrentValue

d-okhrimenko commented on Homework2/CurrentValueCalc.cs:
```diff
@@ -0,0 +1,97 @@
+            PriceOrigin = priceOrigin;
+            PriceTrade = priceTrade;
+            TransactionTypeUser = transactionTypeUser;
+        }
```
Please add extra line betwen constructor and method and remove second extra line after properties list

yasksc committed:
moved try/catch to location where CurrentValueCalc will be used

d-okhrimenko commented on Homework2/Program.cs:
```diff
@@ -10,24 +10,72 @@ class Program
+                        priceOrigin = Convert.ToDouble(Console.ReadLine());
+                        if (priceOrigin < 0)
+                        {
+                            throw new Exception("Original price must be bigger than 0. Try to enter original price one more time:");
```
The previous code with the placement of the throw keyword was correct.
The throw keyword must be in class CurrentValueCalc and try/catch must be in Program. If you can throw and catch in the same file it means that you can rewrite code without exception at all. And code without exceptions is preferred. If you write code for other developers with help of exceptions you can warn them on which way they should use your class. So developing code to give this code to another developer you will use throw meaning that other developers will use catch and will decide how to show an error to the end-user. Also, throw can be used to send messages between layers of application. From business logic to the presentation layer.

alberttashu commented on Homework2/Program.cs:
```diff
@@ -10,6 +10,74 @@ class Program
     {
         static void Main(string[] args)
         {
+            int nominal = 0;
```
It's a good practice to introduce new variables as close to the place where they are used. As example, you use `nominal` first the scope of do-while loop on the 23rd line. So you can declare the variable before just the loop. And the same for all the other variables.

alberttashu commented on Homework2/Program.cs:
```diff
@@ -10,6 +10,74 @@ class Program
+            double priceTrade = 0;
+
+            do
+            {
```
Could you please try to refactor this code to decrease nesting? It's usually hard to read and understand methods that have a lot of conditions/loops/try-catch statements nested one into another. 
You can achieve a better result if you'd move some part of the code into the separate methods (e.g. you can move the code inside the main loop `while (Console.ReadKey().Key != ConsoleKey.Escape)` into the separate method, introduce method to ask user to for positive int input via console)
